torchdet3d/utils/ie_wrappers.py

```diff
@@ -159,3 +161,311 @@ def crop(frame, rect):
+                                                   self.time, self.continue_time_thresh,
+                                                   self.detection_filter_speed, self.keypoints_filter_speed,
+                                                   self.add_treshold, self.no_updated_frames_treshold)
+        return assignment
+
+    def _clear_old_tracks(self):
+        clear_tracks = []
+        for track in self.tracks:
+            # remove too old tracks
+            if track.get_end_time() < self.time - self.track_clear_thresh:
+                self.history_tracks.append(track)
+                continue
+            # remove too short and outdated tracks
+            if track.get_end_time() < self.time - self.continue_time_thresh \
+                    and len(track) < self.time_window:
+                self.global_id_releaser(track.id)
+                continue
+            clear_tracks.append(track)
+        self.tracks = clear_tracks
+
+    def _compute_detections_assignment_cost(self, active_tracks_idx, detections):
+        cost_matrix = np.zeros((len(detections), len(active_tracks_idx)), dtype=np.float32)
+
+        for i, idx in enumerate(active_tracks_idx):
+            track_box = self.tracks[idx].get_last_box()
+            for j, d in enumerate(detections):
+                iou_dist = 0.5 * (1 - self._giou(d, track_box))
+                cost_matrix[j, i] = iou_dist
+
+        return cost_matrix
+
+    def _create_new_tracks(self, detections, kps, assignment):
+        for i, j in enumerate(assignment):
+            if j is None:
+                self.tracks.append(Track(self.global_id_getter(),
+                                         detections[i], kps[i], self.time))
+
+    @staticmethod
+    def _area(bbox):
+        return max((bbox[2] - bbox[0]), 0) * max((bbox[3] - bbox[1]), 0)
+
+    def _giou(self, b1, b2, a1=None, a2=None):
+        if a1 is None:
+            a1 = self._area(b1)
+        if a2 is None:
+            a2 = self._area(b2)
+        intersection = self._area([max(b1[0], b2[0]), max(b1[1], b2[1]),
+                                   min(b1[2], b2[2]), min(b1[3], b2[3])])
+
+        enclosing = self._area([min(b1[0], b2[0]), min(b1[1], b2[1]),
+                                max(b1[2], b2[2]), max(b1[3], b2[3])])
+        u = a1 + a2 - intersection
+        iou = intersection / u if u > 0 else 0
+        giou = iou - (enclosing - u) / enclosing if enclosing > 0 else -1
+        return giou
+
+    def _iou(self, b1, b2, a1=None, a2=None):
+        if a1 is None:
+            a1 = self._area(b1)
+        if a2 is None:
+            a2 = self._area(b2)
+        intersection = self._area([max(b1[0], b2[0]), max(b1[1], b2[1]),
+                                   min(b1[2], b2[2]), min(b1[3], b2[3])])
+
+        u = a1 + a2 - intersection
+        return intersection / u if u > 0 else 0
+
+    def _check_velocity_constraint(self, detection1, det1_time, detection2, det2_time):
```

Comment:
Seems like we can get rid of it: IOU > 0.4 guarantees that object is moving more or less smoothly.